src/block-components/helpers/borders/style.js

```diff
@@ -109,6 +110,27 @@ const Styles = props => {
 				valueCallback={ () => 'hidden' }
 				dependencies={ [ 'overflow' ] }
 			/>
+			<BlockCss
+				{ ...propsToPass }
+				// borderRadius is deprecated, so we had to duplicate the style
+				// generated by the borderRadius to be used by borderRadius2.
```

Comment:
```suggestion
				// borderRadius is deprecated, so we had to duplicate the style
				// generated by the borderRadius to be used by borderRadius2.
				// Adding a border radius should append `overflow: hidden`.
				// This is to prevent gradient background from overflowing.
```